common/common-http/src/main/java/org/ow2/proactive/http/HttpClientBuilder.java

```diff
@@ -0,0 +1,162 @@
+ *    Enterprise Grids & Clouds
+ *
+ * Copyright (C) 1997-2016 INRIA/University of
+ *                 Nice-Sophia Antipolis/ActiveEon
+ * Contact: proactive@ow2.org or contact@activeeon.com
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation; version 3 of
+ * the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+ * USA
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ * ################################################################
+ * $$PROACTIVE_INITIAL_DEV$$
+ */
+package org.ow2.proactive.http;
+
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.SSLContext;
+
+import org.ow2.proactive.web.WebProperties;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+
+public class HttpClientBuilder {
+
+    private static final AllowAllTrustStrategy ACCEPT_ANY_CERTIFICATE_TRUST_STRATEGY = new AllowAllTrustStrategy();
+
+    private Boolean allowAnyHostname;
+
+    private Boolean allowAnyCertificate;
+
+    private int maxConnections;
+
+    private boolean useSystemProperties;
+
+
+    public HttpClientBuilder allowAnyHostname(boolean acceptAnyHostname) {
+        this.allowAnyHostname = acceptAnyHostname;
+        return this;
+    }
+
+    public HttpClientBuilder allowAnyCertificate(boolean acceptAnyCertificate) {
+        this.allowAnyCertificate = acceptAnyCertificate;
+        return this;
+    }
+
+    public HttpClientBuilder insecure(boolean insecure) {
+        allowAnyHostname(insecure);
+        allowAnyCertificate(insecure);
+        return this;
+    }
+
+    public HttpClientBuilder maxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+        return this;
+    }
+
+
+    public HttpClientBuilder useSystemProperties() {
+        return useSystemProperties(true);
+    }
+
+    public HttpClientBuilder useSystemProperties(boolean useSystemProperties) {
+        this.useSystemProperties = useSystemProperties;
+        return this;
+    }
+
+    public CloseableHttpClient build() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "build" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)